File: src/part01/chapter02/levelC/matrix/java/MatrixPrinter.java
package part01.chapter02.levelC.matrix.java;

public class MatrixPrinter {

    public static void printMatrix(double[][] matrix) {
        for (double[] row : matrix) {
            for (double number : row) {
                System.out.print(number + "\t");
            }
            System.out.println();
        }
    }

    public static void printMatrix(long[][] matrix) {
        for (long[] row : matrix) {
            for (long number : row) {
                System.out.print(number + "\t");
            }
            System.out.println();
        }
    }
}

File: src/part01/chapter02/levelC/matrix/kotlin/MatrixPrinter.kt
package part01.chapter02.levelC.matrix.kotlin

fun printMatrix(matrix: Array<DoubleArray>) {
    matrix.forEach { row ->
        row.forEach {
            print("$it\t")
        }
        println()
    }
}

fun printMatrix(matrix: Array<LongArray>) {
    matrix.forEach { row ->
        row.forEach {
            print("$it\t")
        }
        println()
    }
}